drivers/modem/gsm_ppp.c

```diff
@@ -255,6 +256,25 @@ static struct setup_cmd setup_cmds[] = {
 	SETUP_CMD_NOHANDLE("AT+CGDCONT=1,\"IP\",\"" CONFIG_MODEM_GSM_APN "\""),
 };
 
+MODEM_CMD_DEFINE(on_cmd_atcmdinfo_attached)
+{
+	int error = -EAGAIN;
+
+	/* Expected response is "+CGATT: 0|1" so simply look for '1' */
+	if (argc && atoi(argv[0]) == 1) {
```

Comment:
If we are just expecting 0|1 value, then we could avoid the atoi() call by just doing `*argv[0] == '1'`

Reply:
That won't work if `argv[0] = " 1"` for instance.

Reply:
Sure. Ok, lets go with atoi() then if it is not always certain how the values are passed.